src/pykiso/lib/auxiliaries/proxy_auxiliary.py

```diff
@@ -227,31 +227,25 @@ def _run_command(self) -> None:
         """
         for conn in self.proxy_channels:
             if not conn.queue_in.empty():
-                args, kwargs = conn.queue_in.get()
-                message = kwargs.get("msg")
-                if message is not None:
-                    self._dispatch_command(
-                        message=message,
-                        remote_id=kwargs.get("remote_id"),
-                        con_use=conn,
-                    )
-
-                self.channel._cc_send(*args, **kwargs)
-
-    def _dispatch_command(self, message: bytes, con_use: CChannel, remote_id: int):
+                any_msg, raw = conn.queue_in.get()
```

Comment:
If you go for any kind of message you will have to handle the different type in the auxiliary. In addition why keeping the raw, just handle the TestApp decoding in the corresponding auxiliary.

Reply:
That will probably be one of the next steps to do.

Reply:
yep, raw needs to be gone..

Reply:
yep, raw needs to be gone..